text/0001-compsci-bois-almanac.md

```diff
@@ -0,0 +1,115 @@
+- Project / Feature Name: `compsci-bois-almanac`
+- Start Date: 2023-08-17
+- RFC PR: [fogo-sh/rfcs#1](https://github.com/fogo-sh/rfcs/pull/1)
+
+# Summary
+
+[summary]: #summary
+
+A collection of markdown files about different events / things / memes / etc. in the Compsci-Boi-verse, accessible over
+http(s) and gemini, privately.
+
+Images and videos will be supported, but it will be mostly text.
+
+The initial content can be populated from a XML export of the existing Compsci Bois Wiki.
+
+Backlinks will be supported via \[\[this syntax\]\].
+
+After the initial version is released, more syntax can be added to increase more wiki-like functionality, like
+categories / dates / other enriching metadata, along then with pages that can act as indexes that utilize the metadata
+(like a all categories page, or an all posts about events in 2023).
+
+# Motivation
+
+[motivation]: #motivation
+
+Having a location to document different things about our group, from a nostalgic / historical perspective, would be
+nice.
+
+Hopefully, due to the enjoyment of consuming even the existing content of this resource that comes from the existing
+wiki in this new more accessible to programmer format will encourage more people to contribute to it.
+
+["The Space Between Us"](https://goodenough.us/blog/2023-08-18-the-space-between-us/) is a good blog post about the
+creation of content to form 'identity anchors'.
+
+# Drawbacks
+
+[drawbacks]: #drawbacks
+
+We already technically have a wiki solution, [fogo-sh/CSBWiki](https://github.com/fogo-sh/CSBWiki), and MediaWiki is
+quite powerful.
+
+This proposal implies building a bespoke solution, harder than just finding something and running with it.
```

Comment:
We shouldn't discount how powerful it is. The collaboration features are a lot more human compared to writing markdown files and using git and making PRs. The two solutions obviously have different goals and vibes, but I think even if we committed to building the bespoke solution, and then actually built it, we'd be missing solid MediaWiki features that could bring a lot to the experience.

Like imagine we wanted a page about the Grack cryptid, in my mind features like the infobox would be a must-have. Just imagine Grack having its own biographical entry and an infobox:
![image](https://github.com/fogo-sh/rfcs/assets/5544782/80e7ea90-3463-4799-aa67-1fce11709103)

I think that would be funny as shit, and having an actually Wiki seems to fit the vibe better of having a place for "lore" (like the SCP Wiki for example). I think there's a ton of meme potential in infoboxes as well if we refer to certain dramatic incidents using the infobox formatting for a conflict:

![image](https://github.com/fogo-sh/rfcs/assets/5544782/ad5d7b60-0f76-4f7d-ae79-ec05667bbae1)

There's so much comedic value in parodying the format of Wikipedia but for comp sci bois lore that would really make the wiki worthwhile that we wouldn't get with a custom solution.

Lastly, the fact that MediaWiki _isn't_ a static-site generator opens up so many cool possibilities. We can do talk pages, something like https://en.wikipedia.org/wiki/Wikipedia:Teahouse or https://en.wikipedia.org/wiki/Wikipedia:Village_pump.

Far too many good features in MediaWiki to give it up IMO.

Reply:
Couple other ideas in the vein of keeping and rejuvenating the wiki (in addition to a Discord bot handling accounts):

- A famous "quotes" page on the Wiki that can be autopopulated by quotebot.
- A similar "top hits of all time" page that uses starboard posts.

Reply:
I will admit I've fallen out of enjoyment with MediaWiki so while I would also like to replicate aspects of its output from a visual / comedic perspective, I don't want to be involved in:
- Using its editor experience
- Using its antiquated formatting style
- Having to interface with PHP to do anything custom with it
- Needing Riley to beef up his existing MediaWiki setup just so we can access WikiData (a pretty basic feature that afaik is how the infoboxes would even be pulled from)

Reply:
Maybe we should consider building on top of nini in that case? It's so tiny that it provides some of the initial boilerplate and not much else.

Reply:
Using ni / nini as a base would speed things up quite a bit!

Reply:
Might be good to fork the project and consider what pieces we want to take from it though, with regards to what feature we'd like to build out from it, but still considering pumping out a basic MVP quickly as a POC